common/mlsdbserialisation.cpp

```diff
@@ -11,43 +11,70 @@
 */
 
 #include "mlsdbserialisation.h"
+#include "mccmapping.h"
 
-QDataStream &operator<<(QDataStream &out, const MlsdbCoords &coords)
+QString stringForMlsdbCellType(MlsdbCellType type)
 {
-    out << coords.lat << coords.lon;
-    return out;
+    switch (type) {
+        case MLSDB_CELL_TYPE_LTE: return QLatin1String("LTE");
+        case MLSDB_CELL_TYPE_GSM: return QLatin1String("GSM");
+        case MLSDB_CELL_TYPE_UMTS: return QLatin1String("UMTS");
+        default: return QLatin1String("OTHER");
+    }
 }
 
-QDataStream &operator>>(QDataStream &in, MlsdbCoords &coords)
+quint64 getMlsdbUniqueCellId(MlsdbCellType cellType, quint32 cellId, quint32 locationAreaCode, quint16 mcc, quint16 mnc)
 {
-    in >> coords.lat >> coords.lon;
-    return in;
+    quint64 id, temp;
+    quint16 i = 0;
+    // mcc 10 bits, mnc 10 bits, lAC 16 bits, cId 28 bits
+    if (mcc > 1023 || mnc > 1023 || locationAreaCode > 65534 || cellId > 268435455) {
+        fprintf(stderr, "WARNING: Received a value that was too big. mcc was %d, max 1023, mnc was %d, max 1023 locationAreaCode was %d, max 65534, cellId was %d, max 268435455\n", mcc, mnc, locationAreaCode, cellId);
```

Comment:
Cosmetics: quite a long line here, could split to smaller ones. Perhaps even the literal with ""\n""